### PR TITLE
Set `tlJdkRelease := 8` for all modules but `io`

### PR DESCRIPTION
**Pull request by armanbilge**
Although @vasilmkd's release flag idea doesn't work for the io module, there's no reason we shouldn't apply it for all the other modules.

### CODE REVIEW

File: core/shared/src/main/scala/fs2/text.scala [hunk 1]
```diff
@@ -209,7 +209,7 @@ object text {
         }
 
         val isLast = r.isEmpty
-        (lastOutBuffer: Buffer).clear()
+        lastOutBuffer.clear()
```
Comment (armanbilge) [inline core/shared/src/main/scala/fs2/text.scala:212]: This is more about making a point :P we can revert to play it safe.

**Review by vasilmkd — APPROVED** [review]

**Comment by armanbilge** [comment]
In fact, this suggests the solution is to isolate the JDK 16 stuff into its own tiny module, which fs2-io can depend on. Then, we can compile everything but that with this release flag.

**Comment by mpilquist** [comment]
Okay with this PR but not okay with factoring out a new module. Too complex.

File: core/shared/src/main/scala/fs2/text.scala (hunk 1 above)
Comment (mpilquist) [reply core/shared/src/main/scala/fs2/text.scala:212]: Yeah let's revert these and the `flip` ones below. When we need to target 9+ I don't want to have to fix these all over again.

**Pull request merged**